src/karts/kart.cpp

```diff
@@ -1278,9 +1299,22 @@ void Kart::update(float dt)
     {
         m_kart_animation->update(dt);
     }
+
+    time_previous_counter += dt;
+    while (time_previous_counter > (1.0f/120.0f))
+    {
+        m_previous_xyz[0] = getXYZ();
+        for (int i=29;i>0;i--)
```

Comment:
Not so important since it's only 30 elements, but this part could probably be improved a bit by using something like std::deque

Reply:
According to the documentation I just read on the topic ; deque adds insertion to the front and doesn't use contiguous memory.

Neither seem relevant to an array of fixed size without any kind of insertion/deletion (these are just values update which can be done directly, doing the loop in reverse shouldn't affect performance).

Also according to cppreference.com, "deques typically have large minimal memory cost".

Reply:
I guess I was thinking more about ring buffers, that's one common way to implement deques, though seems like the std:: one is more complex than I'd expected. Anyway, since it's only 30 items, it probably doesn't matter much